some systems need segments to be letterboxed to some overarching res

vadosnaprimer opened the pull request:
- target resolution is deduced automatically by looking at all segments
- since we can letterbox at native res during initial import, resizing is not involved
- resize based MS import is incompatible with this setting

`msLetterbox` is the variable that enables this, has to be set in addition to `ms`.

fixed #18

vadosnaprimer committed:
some systems need segments to be letterboxed to some overarching res (wonderswan)

- target resolution is deduced automatically by looking at all segments
- since we can letterbox at native res during initial import, resizing is not involved
- resize based MS import is incompatible with this setting

vadosnaprimer commented:
Tested on https://tasvideos.org/4829M, seems to work fine.

despoa commented:
Few observations and suggestions:
* If the `sample` parameter in `msLetterbox` isn't going to be used at all, it should be removed, unless Avisynth syntax requires it.
* In encode.avs, lines 60 and 61 should be right after line 53 to make it clear to the end user that multisegment letterboxing is an option.
* In the same file, `msAudio` will be completely ignored and will not work if `msLetterbox` is true. Here's a possible fix.
```
if (ms && !msLetterbox) {
	resized = resized \
		.AppendSegment(msBase, msStart, msEnd, msFormat, resizer, hd, pixelType) \
		.ConvertToRGB32()
}

if ( (ms || msLetterbox) && msAudio) {
		resized = AudioDub(resized, WavSource(msAudiotrack))
}
```

vadosnaprimer committed:
- drop the sample arg since it's not used
- movie the option around for more clarity
- set msAudioFile path when there's external audio to import, instead of having both path and bool switch
- allow using external audio for msLetterbox

vadosnaprimer commented:
Check.

despoa commented:
Looks good.

fsvgm777 reviewed (approved):
Works as intended.
